### DISCUSSION TITLE
Temporary URL for Secure Object Sharing with Limited Access Rights(Read only access)

### DISCUSSION THREAD

**Discussion by arian360** [minio/minio#19697] [Q&A] [answered]
Hi Team,

I'm currently seeking a solution to share documents via URL in a read-only format. Could anyone advise on how to accomplish this using Minio?

Thanks!

**Comment by klauspost** [comment]
For single objects you can create a pre-signed url: https://min.io/docs/minio/linux/reference/minio-mc/mc-share.html - valid up to 7 days.

For a prefix you can set up a policy: https://min.io/docs/minio/kubernetes/upstream/administration/identity-access-management/policy-based-access-control.html#userpolicy.readonly

**Reply by arian360** [reply]
"Thanks, @klauspost! Is it feasible to limit access to just viewing the file without allowing downloads using temporary pre-signed URLs?"

**Reply by klauspost** [reply, marked as answer]
"Viewing the file" and "Downloading the file" is the same thing.

**Reply by arian360** [reply]
Got the solution:
https://github.com/minio/minio/issues/19704
